timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -476,28 +477,82 @@ def static_features(self, value: Optional[pd.DataFrame]):
 
         self._static_features = value
 
+    def infer_frequency(self, num_items: Optional[int] = 100, raise_if_irregular: bool = False) -> str:
+        """Infer the time series frequency based on the timestamps of the observations.
+
+        Parameters
+        ----------
+        num_items : int or None, default = 100
+            Number of items (individual time series) randomly selected to infer the frequency. Lower values speed up
+            the method, but increase the chance that some items with invalid frequency are missed by subsampling.
+
+            If set to `None`, all items will be used for inferring the frequency.
+        raise_if_irregular : bool, default = False
+            If True, an exception will be raised if some items have an irregular frequency, or if different items have
+            different frequencies.
+
+        Returns
+        -------
+        freq : str
+            If all time series have a regular frequency, returns a pandas-compatible `frequency alias <https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases>`_.
+
+            If some items have an irregular frequency or if different items have different frequencies, returns string
+            `IRREG`.
+        """
+
+        df = pd.DataFrame(self)
+        if num_items is not None:
+            all_item_ids = self.item_ids
+            if len(all_item_ids) > num_items:
+                items_subset = all_item_ids.to_series().sample(n=num_items, random_state=123)
```

Comment:
Probably overkill but should random_state be also customizable?

Reply:
I don't think a user will ever need to access it. If submsampling with our fixed seed failed to detect problems, the user can run `df.infer_frequency(num_items=None)` to compute the frequency on all items in a deterministic and 100% reliable way. 

Do you think that's reasonable?